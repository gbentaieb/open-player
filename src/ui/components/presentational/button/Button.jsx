import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import { withStyles } from '@material-ui/core';

const styles = {
  root: {
    color: 'white',
  },
};

class Button extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
    Icon: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);
    this.setIsHovered = this.setIsHovered.bind(this);
  }

  state = {
    isHovered: false,
  }

  setIsHovered = (bool) => {
    this.setState(state => ({ ...state, isHovered: bool }));
  }

  render() {
    const {
      onClick,
      classes,
      Icon,
    } = this.props;

    return (
      <div
        className={classes.root}
      >
        <IconButton
          color={this.state.isHovered ? 'primary' : 'inherit'}
          onClick={onClick}
          onMouseEnter={() => this.setIsHovered(true)}
          onMouseLeave={() => this.setIsHovered(false)}
        >
          <Icon fontSize="large" />
        </IconButton>
      </div>
    );
  }
}

export { Button };
export default withStyles(styles)(Button);
