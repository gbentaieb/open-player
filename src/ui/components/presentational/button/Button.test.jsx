import React from 'react';
import { Button } from './Button';

const defaultProps = {
  onClick: () => {},
  glyph: {
    node: {
      innerHTML: '<path d="abcd"></path>',
    },
  },
  classes: {},
};

describe('Button snapshot', () => {
  test('Button > Snapshot', () => {
    const wrapper = shallow(<Button {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Button methods', () => {
  test('Button > setIsHovered', () => {
    const wrapper = shallow(<Button {...defaultProps} />);
    expect(wrapper.state('isHovered')).toEqual(false);
    wrapper.instance().setIsHovered(true);
    expect(wrapper.state('isHovered')).toEqual(true);
  });
});
