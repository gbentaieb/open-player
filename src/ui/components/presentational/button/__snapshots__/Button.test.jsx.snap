// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot Button > Snapshot 1`] = `
<div>
  <WithStyles(IconButton)
    color="inherit"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Icon
      fontSize="large"
    />
  </WithStyles(IconButton)>
</div>
`;
